src/swsssdk/dbconnector.py

```diff
@@ -27,90 +67,134 @@ def load_sonic_db_config(sonic_db_file_path=SONIC_DB_CONFIG_FILE):
-        SonicDBConfig.inst_name_validation(inst_name)
-        return SonicDBConfig._sonic_db_config["INSTANCES"][inst_name]
+        return SonicDBConfig._sonic_db_config.keys()
 
     @staticmethod
-    def get_socket(db_name):
+    def get_instance(db_name, namespace=''):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        SonicDBConfig.db_name_validation(db_name)
-        return SonicDBConfig.get_instance(db_name)["unix_socket_path"]
+        SonicDBConfig.db_name_validation(db_name, namespace)
+        inst_name = SonicDBConfig._sonic_db_config[namespace]["DATABASES"][db_name]["instance"]
+        SonicDBConfig.inst_name_validation(inst_name, namespace)
+        return SonicDBConfig._sonic_db_config[namespace]["INSTANCES"][inst_name]
 
     @staticmethod
-    def get_hostname(db_name):
+    def get_socket(db_name, namespace=''):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        SonicDBConfig.db_name_validation(db_name)
-        return SonicDBConfig.get_instance(db_name)["hostname"]
+        return SonicDBConfig.get_instance(db_name, namespace)["unix_socket_path"]
 
     @staticmethod
-    def get_port(db_name):
+    def get_hostname(db_name, namespace=''):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        SonicDBConfig.db_name_validation(db_name)
-        return SonicDBConfig.get_instance(db_name)["port"]
+        return SonicDBConfig.get_instance(db_name, namespace)["hostname"]
 
     @staticmethod
-    def get_dbid(db_name):
+    def get_port(db_name, namespace=''):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        SonicDBConfig.db_name_validation(db_name)
-        return SonicDBConfig._sonic_db_config["DATABASES"][db_name]["id"]
+        return SonicDBConfig.get_instance(db_name, namespace)["port"]
 
     @staticmethod
-    def get_separator(db_name):
+    def get_dbid(db_name, namespace=''):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        SonicDBConfig.db_name_validation(db_name)
-        return SonicDBConfig._sonic_db_config["DATABASES"][db_name]["separator"]
+        SonicDBConfig.db_name_validation(db_name, namespace)
+        return SonicDBConfig._sonic_db_config[namespace]["DATABASES"][db_name]["id"]
+
+    @staticmethod
+    def get_separator(db_name, namespace=''):
+        if SonicDBConfig._sonic_db_config_init == False:
+            SonicDBConfig.load_sonic_db_config()
+        SonicDBConfig.db_name_validation(db_name, namespace)
+        return SonicDBConfig._sonic_db_config[namespace]["DATABASES"][db_name]["separator"]
 
 class SonicV2Connector(DBInterface):
-    def __init__(self, use_unix_socket_path=False, **kwargs):
+    def __init__(self, use_unix_socket_path=True, **kwargs):
```

Comment:
>True [](start = 44, length = 4)

Change default behavior will break backward compatibility.

Reply:
The default behavior was originally use_unix_socket_path=True, but was reverted in this commit. Do you remember any specific reason Qi ?

commit bc3964b788c3a4a45f2b359a5df5934ecdee84c2
Author: Dong Zhang <41927498+dzhangalibaba@users.noreply.github.com>
Date:   Tue Nov 5 09:13:05 2019 -0800

I remember originally unix socket was used as default as it was faster than TCP socket.

Reply:
I did try loading the image in the DUT and it was coming up fine.
But since you are echoing issues you faced earlier setting default as UNIX_socket , I would revert back to default use_unix_socket_path=False.